src/sdk/bucketspace_fs.js

```diff
@@ -186,18 +194,38 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
     }
 
     /**
-     * check_same_stat will return true the config file was not changed
+     * check_same_stat_bucket will return true the config file was not changed
+     * @param {string} bucket_name
      * @param {nb.NativeFSStats} bucket_stat
      * @returns Promise<{boolean>}
      */
-    async check_same_stat(bucket_name, bucket_stat) {
+    async check_same_stat_bucket(bucket_name, bucket_stat) {
         try {
             const current_stat = await this.config_fs.stat_bucket_config_file(bucket_name);
             if (current_stat) {
                 return current_stat.ino === bucket_stat.ino && current_stat.mtimeNsBigint === bucket_stat.mtimeNsBigint;
             }
         } catch (err) {
-            dbg.warn('check_same_stat: current_stat got an error', err, 'ignoring...');
+            dbg.warn('check_same_stat_bucket: current_stat got an error', err, 'ignoring...');
+        }
+    }
+
+    /**
+     * check_same_stat_bucket will return true the config file was not changed
+     * in case we had any issue (for example error during stat) the returned value will be undefined
+     * @param {string} account_id
+     * @param {nb.NativeFSStats} account_stat
+     * @returns Promise<{boolean|undefined>}
+     */
+    async check_same_stat_account(account_id, account_stat) {
```

Comment:
why check_same_stat_account is in bucketspace_fs and not in accountspace_fs?

Reply:
I think it is related to another comment you raised ([here](https://github.com/noobaa/noobaa-core/pull/8585#discussion_r1888280651)).
 Because I know I have bucketspace in all the services, the function of `read_account_by_access_key` in NC is implemented inside bucketspace_fs.

Reply:
having `read_account_by_access_key` in the bucketspaces was the desired behavior as long as we didn't have accountspaceFS.
Now that we do, it doesn't make sense anymore to have additional account-related functionalities in the bucketspaces.
Same for the account cache, why it's in objectSDK and not in accountSDK? 
Account cache move is not something that we can easily change, especially not in containerized env where we currently don't have accountSDK. @guymguym WDYT?
@shirady these changes are not in the right place, validate_account should call accountspaces functionalities.

Reply:
I think you're right, and while trying to assess the change of moving all the related function to account_sdk and accountspace seems not trivial, I mentioned this as a gap.